pkg/ingester/ingester_v2.go

```diff
@@ -216,6 +242,37 @@ func (u *userTSDB) setLastUpdate(t time.Time) {
 	u.lastUpdate.Store(t.Unix())
 }
 
+func (u *userTSDB) canIdleClose(idleTimeout time.Duration) (bool, error) {
+	if !u.isIdle(time.Now(), idleTimeout) {
```

Comment:
I just realised this check has a problem: the `lastUpdate` is not correct after an ingester restart. An option could be checking the timestamp against `max(lastUpdate, TSDB max time)`.

Reply:
When TSDB is opened, we set `lastUpdate` to current time to avoid premature idle flush and close. This will only be a problem if ingesters never stay up for longer than configured idle timeout.

Reply:
Instead of current time, we can set it to last sample time from TSDB when opening, as you suggest.

Reply:
> When TSDB is opened, we set lastUpdate to current time to avoid premature idle flush and close. 

Then we're protected 👍 

> Instead of current time, we can set it to last sample time from TSDB when opening, as you suggest.

Could make sense.